### DISCUSSION TITLE
Awful results since skip text encoder was removed.

### DISCUSSION THREAD

**Discussion by metzo007** [TheLastBen/fast-stable-diffusion#1066] [General]
Hi,

since the option to remove the text encoder was removed, I am not able to get any good results. 

Today I did a training with the min 350 Text encoder training steps and the model and I got no dreambooth at all. The results are like using the 1.5 model only, 

Yesterday I did the same with 600 steps and the person fron the trainen pictures got 3 arms and the posture was all over the place. 

I also did it with 100% UNet and it ignore all my prompts completely
but with 100% Text encoder it always a mix of the training images

I used the same dataset and training steps that I used before the change, and while the old model is very good, all the models I trained after the change are awful. 
I just try to get the same results as before, but sadly the old versions don't work anymore.  
Anyone can help?

**Comment by metzo007** [comment]
One more thing to add. We had an option to skip the text encode before and it worked fine, but if we now skip the text encoder by set it to zero, SD will ignore the prompts. So there was different method before. What was it?

**Reply by TheLastBen** [reply]
alway train the text encoder for at least 300 steps, this is how it learns your instance name

**Reply by nawnie** [reply]
> One more thing to add. We had an option to skip the text encode before and it worked fine, but if we now skip the text encoder by set it to zero, SD will ignore the prompts. So there was different method before. What was it?

sd will ignore the prompts, do you have style checked, cause that's what style training does.
for me i use style after the image looks close, at that point the model knows the tags/caption so only training the images itself helps it get a better grasp w/o over fitting.

**Reply by metzo007** [reply]
@TheLastBen This is my problem. I did train for the text encoder for 300 steps and got no effect at all. Than I did train text encoder only and got a overfitting. It is a guessing game now.

@nawnie. I am training on a person. I can try style to see if I get better results. 

Question remaines. How do I skip the text encoler like before when there was a option to skip it?

**Reply by TheLastBen** [reply]
just set it to 0 and it will not train it, because "skip text encoder was not removed" it's a very simple deduction when you set the number of steps to 0 it means no training of the text encoder

**Reply by metzo007** [reply]
But the resulds are not the same. With "skip text encoder" the resulting model was very responsive to any prompt. If I set the text encoder to 0 the resulting model will behave like a stock 1.5 model without any training. 

I only did one test. If you say "skip text encoder" is the same like Unet Training only now, I will test again.

**Reply by TheLastBen** [reply]
Even previously when you don't train the text encoder you will get vanilla results

**Reply by metzo007** [reply]
vanilla results = stock 1.5 model and not responsive to prompts? Is that what you are saying?

I alway skipped the text encoder and got a model that was very responsive to the prompt. Also, if i remebber correctly, there was a time when the text encoder was not even an option, and I got good results. This it what confuses me the most.

**Reply by TheLastBen** [reply]
you can't skip the text encoder and still get results, unless you caption the instance images

**Reply by metzo007** [reply]
caption the instance images = phtmejhn (1).jpg, phtmejhn (2).png.....

Yes, I did name them like this and skipped the text encoder. I am now confused on what the text encoder does? Is the encoder getting description for the images and train with the description. 
If this is true,  caption the instance images and using Unet training only should get the resulds I want. I will try later.

**Reply by nawnie** [reply]
A caption is giving the image a sentence/prompt instead of a single word name.


Phtmejhn is not your caption, that's your instance/subject word. "a photo of phtmejhn going to the store" is a caption. Now with no caption the text encoder would see the image and read it as, "phtmejhn-a person walking outside towards a building"

Do you see why both are not needed, at least in the context of this dreambooth.

**Comment by nawnie** [comment]
Hey ben, iirc there was a point where you had the text encoder train for a
few steps even when at zero back when we were using percents a few months
ago. This may be why this user is having the issues now.

**Reply by TheLastBen** [reply]
Probably, but back then, when text_enc gets disabled, the training would run without the text_enc flag

**Comment by leonnn1** [comment]
I am getting some great results that I couldn't reach before, I am liking this model, it's just more complicated to get the grip of I think
